src/recipe/admin/src/runtime/components/DataTable.vue

```diff
@@ -100,4 +124,18 @@ const value = computed(() =>
 );
 const footerColSpan = computed(() => columns.length - footerTemplate?.columns.length);
 const scrollable = scrollHeight !== undefined;
+const formatter = ref();
+
+function exportDataTable() {
+  composableResolver.resolve(exportOptions.formatter).then(result => {
+    formatter.value = result.default();
+    datatable.value.exportCSV();
+  });
+}
+
+function exportFunction({ data, field }) {
+  if(!formatter.value || !formatter.value) { return data; }
+
+  return formatter.value.format(data, field);;
+}
```

Comment:
i'm not sure what it is, but there is a bad smell here. i will go into detail in the next review